RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -637,6 +636,10 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
      }
    }
 
+   // muon arbitration
+   fillArbitrationInfo( outputMuons.get() );
+   arbitrateMuons( outputMuons.get(), caloMuons.get() );
```

Comment:
IMO, it's easier to add an option and apply arbitration just to the default ```_muons_``` than to prove there are no issues and no cost in all alternative reconstructions.
But we can go the other way and then this PR needs tests for all of these cases (cosmics, seeding, displaced muons).